pyteal/compiler/optimizer/optimizer_test.py

```diff
@@ -0,0 +1,417 @@
+from .optimizer import OptimizeOptions
+
+from pyteal.ir.ops import Op
+
+from pyteal import *
+
+import pytest
+
+
+def test_optimize_subroutine():
```

Comment:
Could you also add a version of this test that uses a subroutine with a `ScratchVar` argument? I think there will still be things to optimize in that case.